File: DeliveryOrderService/src/delivery-order/interfaces/idelivery-order.service.ts
import { OrderStatus } from '../enums/order-status.enum';
import { DeliveryOrder } from '../schemas/delivery-order.schema';
import { Invoice } from '../schemas/invoice.schema';

export interface IDeliveryOrderService {
  store(order: DeliveryOrder): Promise<number>;
  select(supplierId: number): Promise<DeliveryOrder>;
  update(orderId: number, order: Partial<DeliveryOrder>): Promise<OrderStatus>;
  orderShipped(orderId: number): Promise<OrderStatus>;
  confirmPayment(orderId: number, amount: number): Promise<OrderStatus>;
  read(orderId: number): Promise<DeliveryOrder>;
  requestInvoice(orderId: number): Promise<Invoice>;
  prepareToCommitOrder(orderId: number): Promise<DeliveryOrder>;
  commit(orderId: number): Promise<void>;
  abort(orderId: number): Promise<void>;
}

File: Broker/BrokerNCoordinator/src/main/java/com/example/BrokerNCoordinator/Service/Interface/IDeliveryOrderService.java
package com.example.BrokerNCoordinator.Service.Interface;
import com.example.BrokerNCoordinator.DTO.Invoice;
import com.example.BrokerNCoordinator.DataModel.DeliveryOrder;
import com.example.BrokerNCoordinator.DataModel.Enum.OrderStatusType;

public interface IDeliveryOrderService {

    Integer store(DeliveryOrder order);

    DeliveryOrder select(Integer supplierId);

    OrderStatusType update(Integer orderId, DeliveryOrder order);

    OrderStatusType orderShipped(Integer orderId);

    OrderStatusType confirmPayment(Integer orderId, Double amount);

    DeliveryOrder read(Integer orderId);

    Invoice requestInvoice(Integer orderId);

    DeliveryOrder prepareToCommitOrder(Integer orderId);

    void commit(Integer orderId);

    void abort(Integer orderId);
}
